Source/WebKit/NetworkProcess/NetworkProcess.cpp

```diff
@@ -2249,7 +2250,7 @@ void NetworkProcess::prepareToSuspend(bool isSuspensionImminent, MonotonicTime e
     RefPtr<CallbackAggregator> callbackAggregator = CallbackAggregator::create([this, completionHandler = WTFMove(completionHandler)]() mutable {
         RELEASE_LOG(ProcessSuspension, "%p - NetworkProcess::prepareToSuspend() Process is ready to suspend", this);
         UNUSED_VARIABLE(this);
-        completionHandler();
+        completionHandler(memoryFootprint());
```

Comment:
This looks very weird. We only make use of this information for the WebProcesses and yet require all child processes to compute it and send it back, at a time where we want the process to suspend ASAP.

Reply:
Thanks for reviewing, Chris! I will address these issues in a follow-up patch :)